### PR TITLE
Implement Curve Origin[sc-2667]

### PR DESCRIPTION
**Pull request by deep-quality-dev**
### description of pull request:

Implemented Curve origin & Unit Test

### CODE REVIEW

File: config_next.hcl [hunk 1]
```diff
@@ -1,12 +1,66 @@
 # Test config for the gofernext and ghostnext apps. Not ready for production use.
 
+ethereum {
```
Comment (deep-quality-dev) [inline config_next.hcl:3]: Copied from `config.hcl`.

File: pkg/config/dataprovider/origin.go [hunk 2]
```diff
@@ -33,6 +33,19 @@ type configOriginTickGenericJQ struct {
 	JQ  string `hcl:"jq"`
 }
 
+type configContracts struct {
```
Comment (deep-quality-dev) [inline pkg/config/dataprovider/origin.go:36]: Copied the structure from [balancer v2 branch](https://github.com/chronicleprotocol/oracle-suite/blob/sc-2666-balancer/pkg/config/dataprovider/origin.go#L36).

File: pkg/datapoint/origin/origin.go [hunk 3]
```diff
@@ -19,3 +23,37 @@ type Origin interface {
 	// must verify returned data.
 	FetchDataPoints(ctx context.Context, query []any) (map[any]datapoint.Point, error)
 }
+
+func queryToPairs(query []any) ([]value.Pair, bool) {
```
Comment (deep-quality-dev) [inline pkg/datapoint/origin/origin.go:27]: Copied this file from [balancer v2 branch](https://github.com/chronicleprotocol/oracle-suite/blob/sc-2666-balancer/pkg/datapoint/origin/origin.go)

File: config_next.hcl (hunk 1 above)
Comment (deep-quality-dev) [reply config_next.hcl:3]: Should be updated after PR#171.

**Review by teghnet — APPROVED** [review]

**Pull request merged**